dz14_ex1.py

```diff
@@ -0,0 +1,51 @@
+import json
+import re
+
+try:
+    with open('telbook.json') as dfile:
+        telefon_book = json.load(dfile)
+except FileNotFoundError:
+    telefon_book = {}
+
+#print (telefon_book)
+
+while True:
+    command_input = input(' Введіть команду (stats, add, delete, list, show):  ')
+    if command_input == 'stats':
+        print('Кількість записів', len(telefon_book))
+    elif command_input == 'add':
+        new_name = input("введіть ім'я нового запису: ")
+        if telefon_book.get(new_name) is not None:
+            print(' Запису під іменем ', new_name, 'у телефонній книзі існує. ')
+            print(' Ви не можете змінити номер. Рекомендую видалити запис та додати новий.')
+        else:
+            errornomber = True
+            while errornomber:
+                new_fone = input("введіть номер телефону: ")
+                if re.findall(r'\+380\d{9}$|380\d{9}$|0\d{9}$', new_fone):
```

Comment:
Your pattern allows a lot of invalid formats:
```
l+380951234567 -> VALID
phone380953985657 -> VALID
+830951234567 -> VALID
phone380951234567 -> VALID
a0951234567 -> VALID
8+30951234567 -> VALID
a+380951234567 -> VALID
80951234567 -> VALID
+830965254141 -> VALID

Reply:
number verification flaws have been fixed.